stix_shifter_modules/cybereason/stix_transmission/api_client.py

```diff
@@ -0,0 +1,68 @@
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils import logger
+
+DEFAULT_LIMIT = 10000
+DEFAULT_OFFSET = 0
+
+
+class APIClient:
+    LOGIN_ENDPOINT = 'login.html'
+    QUERY_ENDPOINT = "rest/visualsearch/query/simple"
+    LOGOFF_ENDPOINT = 'logout'
+
+    def __init__(self, connection, configuration):
+        self.logger = logger.set_logger(__name__)
+        headers = {'Content-Type': 'application/json'}
+        url_modifier_function = None
+        self.auth = configuration.get('auth')
+        self.client = RestApiClient(connection.get('host'),
+                                    connection.get('port', None),
+                                    headers,
+                                    url_modifier_function=url_modifier_function,
+                                    cert_verify=connection.get('selfSignedCert', True),
+                                    sni=connection.get('sni', None)
+                                    )
+        self.timeout = connection['options'].get('timeout')
+
+    def ping_box(self):
+        """
+        Ping the Data Source
+        :return: Response object
+        """
+        headers = {}
+
+        query = '{"queryPath": [{"requestedType": "Connection", "filters": [], "isResult": true}],' \
+                '"totalResultLimit": 1, "perGroupLimit": 1,"templateContext": "SPECIFIC"}'
+        headers['Cookie'] = self.get_cookies_id()
+        return self.client.call_api(self.QUERY_ENDPOINT, 'POST', headers=headers, data=query)
+
+    def get_search_results(self, query):
+        """
+        Get results from Data Source
+        :param query: Data Source Query
+        :return: Response Object
+        """
+        headers = {'Cookie': self.get_cookies_id()}
+        self.logger.debug("query: %s", query)
+        return self.client.call_api(self.QUERY_ENDPOINT, 'POST', headers=headers, data=query)
+
+    def get_cookies_id(self):
+        """
+        Get session cookie id
+        :return: str, cookie id
+        """
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        response_wrapper = self.client.call_api(self.LOGIN_ENDPOINT, 'POST', headers=headers, data=self.auth)
+        cookie_id = response_wrapper.response.request.headers
+        return cookie_id['Cookie']
```

Comment:
instead of getting cookies id. you can create a `session_log_in(self)` function to query the login endpoint and use the response for ping and results calls.